### PR TITLE
SCSS: Fix broken comments between declaration prop and value

### LINKED ISSUE #5603: SCSS single-line comments are formatted incorrectly
**Important Note:**
Prettier has nice support for Grid-Template-Areas . See [here](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEA6A5gJwJYBMAEwAOlPvlngLTwC2ADgDYCG8lTmcTAzkiWWURAALTrjiYyIpmIn4pMwX36CueOACN2+SLAQxt0eLEWllIAGYQI8WZeviydm4IDcJAL4gANCAh0Y2NBcyKDsmBAA7gAK7AjBKEwAbhB43iDqmExgANZwMADKdFnYUOjIMJgArnA+QjA0DADqQtjwXEVgcPlxrdiJrQCeyOBcwT4lXOIwUZnoNEzI5kwMkz4AVlwAHgBCmTl5+Uw0cAAyJXCLy6sgG5v5JegMcACKlfaXKzUgRZiTmMNcMCjNJ0HCwRp4GBCZAADgADD5QRBJo1MnRhqC4H9EhcfBwAI6VbAcGZMOYLJBLT4+SY0bDlKpfVSlJ6vd6Uq5fGBMdQQ3BQ5AAFh8FSY2AYDwAwhAaPNhlBoLiQJVJgAVHnxKnXbgwD7XXAQMB6r4QSowOhmgBMxvc7iAA )

The example shows only how to reproduce the issue with single-line comments!

**Prettier 1.15.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6VACAKgCwJYDOGhGAygMKmkB0GACgE5wwx5wMYC2EAbnETBxwMAQyYiMBAA5wweAGZ4wI1tAwBXKRgBGcfFAAmGQcPQZInTghgAdKNQDmDPEeB2MHzx6cuAtPE4pABsVOF8xOBECJAx0dy8EmxAhEQN2DBS0jkz2JNjUeITPJIIXOG0xc2h4WCrYazy4qCKvJPkICHgOds70nq6kgG53AF8QABoQCClVKAJkUDEGCAB3Ogi55BARHggXCZBtBhEwAGtmUikTvCgHZBgGdThJnBhOIIB1fHhpE7hSBClVg8PAwACeWzABHmkxuBHYMEYIgcnBEyHkIiC8MmACsCAAPABCxzOFxEVgAMjc4OjMdiQHj8aQbg4gnAAIrqXq0rHPEBXBjwhhbAhQmH85ywD4uQTIAAcAAZJlJlvCPscpFsVfx2HwDkwAI7qPBMJEotFIDG8ybwzh4e6PPmlW5szncy10vkwETaaUGWVIABMkweIjwQRZ5AglgtICg0Bpk3U8KwPvmHt5IxGQA)
```sh
--parser scss
```

**Input:**
```scss
// This is SCSS. Prettier moves the area specification up behind the // comment
.grid {
        grid-template-areas: //
          "header header header" //
          "sidebar content content" //
          "footer footer footer";
 }
```

**Output:**
```scss
// This is SCSS. Prettier moves the area specification up behind the // comment
.grid {
  grid-template-areas: // "header header header"//
    "sidebar content content"//
    "footer footer footer";
}
```

**Expected behavior:**
```scss
// This is SCSS. Prettier moves the area specification up behind the // comment
.grid {
  grid-template-areas: // 
    "header header header" //
    "sidebar content content" //
    "footer footer footer";
}
```

There are related (closed) Issues for older Versions:
- https://github.com/prettier/prettier/issues/4311
- https://github.com/prettier/prettier/issues/4594

### LINKED ISSUE #7109: Broken SCSS Comment Fixing
**Prettier 1.19.1**

```sh
# Options (if any):
--parser postcss --print-width 120 --single-quote
```

**Input:**
```scss
.test {
    background:
        /////// foo
        // bar

        radial-gradient(circle farthest-corner at 5% 10%, #000000, transparent 50%);
}

```

**Output:**
```scss
.test {
    background:
        /////// foo
        // bar radial-gradient(
        circle farthest-corner at 5% 10%,
        #000000,
        transparent 50%
    );
}

```

**Expected behavior:**
```scss
.test {
    background:
        /////// foo
        // bar

        radial-gradient(
            circle farthest-corner at 5% 10%,
            #000000,
            transparent 50%
        );
}

```

### LINKED ISSUE #8052: [SCSS] rule gets accidentally commented out
**Prettier 2.0.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEASAZtGBadBDAWwEsAbAT2wCciwALJAHSgAJmB6N5sAVwGcYIBJqwDkAOQgDmAZTxRezACoBhEQBpm4yRBlyF01Rq1TZ85gCkACuubD2nWfmrNMlZgHlpzABrM5AE2YiT2YAClNmADFKOTAiXkgASjtsPAAHNJI4bF4yfjgCDQAhEiIoAGsAWTwwaTz4AkisNTsOFzwSEgAjGvK7ACUILu0jAAk4EgA3OBgaPGYxOG44G3GpmbmNAEFqDo1ePRy4anQWlntmAogAKyI7ES2MrOZlCBIINwBRAhuiGxFpHAAOYQODMACqAElmN9fv9ASCwVCZGQCMMSCJmABCfxwfDcEgwADcTBAahAEDSs2gvGQoDwlEoEAA7pYGQhaSgOsy8HkySAujEwOUZtI0jUykDkDBKMtybQYAQSAB1WhEeC8cVgOCA+TqoiTdVkZDgXi08llXjHGCWGJAgh4ZD4EhW8nXXgADyKQpFMFkBDgABkynAnR1XSB3R7pJKsgBFbiSUNIZ0R8WUK2UE0JM38tLUWDKoj+GC0ZAADgADOT8xArcqYmkTfm4JnpvyAI6J+C2ymckB4XjYKBwOC4-z8yhwLtEKe2vD2x0p8NwclW4jS2WrkC8WNwBNJsMu7cwPBdIslstIABM5JleFIkteBAdJtbAFZ+Xw4Ioz5zU9ukzLJCUC4rA0hgNQVJbKB0gwGQWRHlaAC+yFAA)

**Input:**

```scss
$font-family-rich:
  // custom
  'Noto Sans TC', 'Noto Sans SC', 'Noto Sans JP', 
  // Safari for OS X and iOS (San Francisco)
  -apple-system, BlinkMacSystemFont,
  // fallback
  Roboto, 'Helvetica Neue', Helvetica, Arial, sans-serif,
  // emoji
  'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol' !default;
```

**Output:**

```scss
$font-family-rich:
  // custom "Noto Sans TC", "Noto Sans SC", "Noto Sans JP",
  // Safari for OS X and iOS (San Francisco)
    -apple-system,
  BlinkMacSystemFont,
  // fallback
    Roboto,
  "Helvetica Neue", Helvetica, Arial, sans-serif,
  // emoji
    "Apple Color Emoji",
  "Segoe UI Emoji", "Segoe UI Symbol" !default;
```

**Expected behavior:**

No change. Or, no behavior changed at least.

Note that `"Noto Sans TC", "Noto Sans SC", "Noto Sans JP",` is accidentally commented after prettier.

```scss
$font-family-rich:
  // custom
  'Noto Sans TC', 'Noto Sans SC', 'Noto Sans JP', 
  // Safari for OS X and iOS (San Francisco)
  -apple-system, BlinkMacSystemFont,
  // fallback
  Roboto, 'Helvetica Neue', Helvetica, Arial, sans-serif,
  // emoji
  'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol' !default;
```

---

Maybe related to https://github.com/prettier/prettier/issues/8015

### PR DESCRIPTION
**Pull request by fisker**
Fixes #5603
Fixes #8052
Fixes #7109

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
I forgot mark this ready, this not prefect, but prints better.

File: tests/scss/comments/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -123,6 +123,130 @@ printWidth: 80
...
+  "Noto Sans TC", "Noto Sans SC", "Noto Sans JP",
+  // Safari for OS X and iOS (San Francisco)
+    -apple-system,
+  BlinkMacSystemFont,
+  // fallback
+    Roboto,
+  "Helvetica Neue", Helvetica, Arial, sans-serif,
+  // emoji
+    "Apple Color Emoji",
+  "Segoe UI Emoji", "Segoe UI Symbol" !default;
```
Comment (fisker) [inline tests/scss/comments/__snapshots__/jsfmt.spec.js.snap:226]: I know this still ugly, but stable even worse

**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEASAZtGBadBDAWwEsAbAT2wCciwALJAHSgAJmB6N5sAVwGcYIBJqwDkAOQgDmAZTxRezACoBhEQBpm4yRBlyF01Rq1TZ85gCkACuubD2nWfmrNMlZgHlpzABrM5AE2YiT2YAClNmADFKOTAiXkgASjtsPAAHNJI4bF4yfjgCDQAhEiIoAGsAWTwwaTz4AkisNTsOFzwSEgAjGvK7ACUILu0jAAk4EgA3OBgaPGYxOG44G3GpmbmNAEFqDo1ePRy4anQWlntmAogAKyI7ES2MrOZlCBIINwBRAhuiGxFpHAAOYQODMACqAElmN9fv9ASCwVCZGQCMMSCJmABCfxwfDcEgwADcIDUIAgaVm0F4yFAeEolAgAHdLPSEDSUB0mXg8qSQF0YmByjNpGkamUgcgYJRlmTaDACCQAOq0IjwXhisBwQHyNVESZqsjIcC8Glksq8Y4wSwxIEEPDIfAkS1k668AAeRUFwpgsgIcAAMmU4I6Oi6QG73dIJVkAIrcSQhpBO8NiyiWyjGhKmvlpaiwJVEfwwWjIAAcAAYyXmIJalTE0sa83AM9M+QBHBPwG0UjkgPC8bBQOBwXH+PmUOCdoiTm14O0O5NhuBky3EKUylcgXgxuDxxOh51bmB4LqF4ulpAAJjJ0rwpAlrwI9uNLYArHy+HBFKeOSmt5MyyQlAuKwNIYDUJSWwgdIMBkFkh6WgAvkhQA)
```sh
--parser scss
```

**Input:**
```scss
$font-family-rich:
  // custom
  'Noto Sans TC', 'Noto Sans SC', 'Noto Sans JP', 
  // Safari for OS X and iOS (San Francisco)
  -apple-system, BlinkMacSystemFont,
  // fallback
  Roboto, 'Helvetica Neue', Helvetica, Arial, sans-serif,
  // emoji
  'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol' !default;
```

**Output:**
```scss
$font-family-rich:
  // custom "Noto Sans TC", "Noto Sans SC", "Noto Sans JP",
  // Safari for OS X and iOS (San Francisco)
    -apple-system,
  BlinkMacSystemFont,
  // fallback
    Roboto,
  "Helvetica Neue", Helvetica, Arial, sans-serif,
  // emoji
    "Apple Color Emoji",
  "Segoe UI Emoji", "Segoe UI Symbol" !default;

```

**Comment by evilebottnawi** [comment]
We can improve it in the future, broken code is worse than ugly

**Comment by fisker** [comment]
Forgot changlog, will add tomorrow.

**Pull request merged**